### PR TITLE
Migrate Panel to Bootstrap 4.1.x

### PR DESCRIPTION
**Pull request by Chng-Zhi-Xuan**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Other, migration

Part of MarkBind/markbind#298

**What changes did you make? (Give an overview)**
- Refactor `panel` tags into `card` tags, as `panel` is deprecated in Bootstrap 4
- Added additional computed properties and in-line template logic to account for syntax change in Bootstrap 4.
- Update button styling and made it dynamic W.R.T the panel's background colour.

Note |
--- |
`Glyphicons` are deprecated in Bootstrap 4. Therefore, buttons in Panel that use the icons are not rendered. The fix for this will be in a separate PR in the MarkBind repository. |
Even though `panel` tags have been refactored to `card`, the Vue component still uses `<panel>` as its name. The author can continue using `<panel>` without refactoring to `<card>` yet.
The user guide has to be updated as `default` type is not supported anymore. There are also other new types / styles available, such as `dark`.

**Testing instructions:**
1. Download [testPanelsV3.zip](https://github.com/MarkBind/vue-strap/files/2154208/testPanelsV3.zip)
1. Run `npm run build` and copy over the new `vue.min.js` and `vue-strap.min.js` to your MarkBind repository folder.
1. Run `markbind serve` in the `PanelTest` folder, provided above.
1. Test the provided panels.

### CODE REVIEW

File: src/Panel.vue [hunk 1]
```diff
@@ -30,7 +28,7 @@
                                       @click.native.stop.prevent="expand()"
                                       @is-open-event="retrieveOnOpen"></panel-switch>
                         <button type="button" class="close-button btn btn-default"
```
Comment (yamgent) [inline src/Panel.vue]: Migration guide suggest that we change `btn-default` to `btn-secondary`. This change is applicable to `PanelSwitch.vue` as well.

File: src/Panel.vue [hunk 2]
```diff
@@ -1,27 +1,25 @@
...
+            </button>
         </div>
-        <div :class="['panel', panelType, {'expandable-panel': isExpandablePanel}]" v-show="!localMinimized">
-            <div :class="['panel-heading',{'accordion-toggle':canCollapse}]"
+        <div :class="['card', {'expandable-card': isExpandableCard}, borderType]" v-show="!localMinimized">
+            <div :class="['card-header',{'accordion-toggle':canCollapse}, cardType, borderType]"
                  @click.prevent.stop="canCollapse && toggle()"
                  @mouseover="onHeaderHover = true" @mouseleave="onHeaderHover = false">
                 <div class="header-wrapper">
                     <span :class="['caret', {'caret-collapse': !localExpanded}]" v-show="showCaret"></span>
```
Comment (yamgent) [inline src/Panel.vue]: Will this be eventually be replaced by a glyphicon? Or is there a unicode equivalent that we can use for the caret?

Comment (yamgent) [reply src/Panel.vue]: Update: While porting Dropdown, I saw this in the migration guide:

> Dropdown toggles no longer require an explicit `<span class="caret"></span>;` this is now provided automatically via CSS’s `::after` on `.dropdown-toggle`.

I checked the CSS, but couldn't find out how they render the caret through the `::after` selector. The content is `""` yet it seems to render the caret somehow? Seems kinda magical 🤔. Will let you know if I found anything new.

**Comment by Chng-Zhi-Xuan** [comment]
**Update**
- Update PR description
- Apply changes requested
- Fix styling of `btn-default`

File: src/Panel.vue [hunk 3]
```diff
@@ -26,13 +26,13 @@
                     <slot name="button">
                         <panel-switch v-show="canCollapse && !noSwitchBool && !showCaret" v-bind:is-open="localExpanded"
                                       @click.native.stop.prevent="expand()"
-                                      @is-open-event="retrieveOnOpen"></panel-switch>
-                        <button type="button" class="close-button btn btn-default"
+                                      @is-open-event="retrieveOnOpen" v-bind:is-light-bg="isLightBg"></panel-switch>
```
Comment (yamgent) [inline src/Panel.vue]: ~~v-bind~~:is-light-bg="isLightBg"

File: src/Panel.vue [hunk 4]
```diff
@@ -1,59 +1,57 @@
...
                     </slot>
                 </div>
                 <div class="button-wrapper">
                     <slot name="button">
                         <panel-switch v-show="canCollapse && !noSwitchBool && !showCaret" v-bind:is-open="localExpanded"
                                       @click.native.stop.prevent="expand()"
-                                      @is-open-event="retrieveOnOpen"></panel-switch>
-                        <button type="button" class="close-button btn btn-default"
-                                v-show="this.type !== 'seamless' ? (!noCloseBool) : onHeaderHover"
+                                      @is-open-event="retrieveOnOpen" v-bind:is-light-bg="isLightBg"></panel-switch>
```
Comment (yamgent) [inline src/Panel.vue]: ~~v-bind~~:is-light-bg="isLightBg"

File: src/Panel.vue [hunk 5]
```diff
@@ -1,59 +1,57 @@
...
-                    <template v-else>
-                        <slot name="header">
-                            <div class="panel-title" v-html="altContent"></div>
-                        </slot>
-                    </template>
-                </button>
-            </div>
+            <button :class="['morph-display-wrapper', 'btn', btnType, 'card-title']" @click="open()">
+                <template v-if="altContent">
+                    <div  v-html="altContent"></div>
```
Comment (yamgent) [inline src/Panel.vue]: Remove additional spacing between 'div' and 'v-html'.

**Comment by Chng-Zhi-Xuan** [comment]
**Update**
- Fix small typo
- Remove or refactor traces of accordion logic within `Panel.vue`

**Pull request merged**